lib/metadata.js

```diff
@@ -204,9 +205,15 @@ Metadata.prototype.generateOidcPEM = function generateOidcPEM(kid) {
     const exponent = new Buffer(key.e, 'base64');
 
     pubKey = aadutils.rsaPublicKeyPem(modulus, exponent);
-    return pubKey;
+    foundKey = true;
+    return true;
   });
 
+  if (!foundKey)
+    throw new Error(`a key with kid %s cannot be found`, kid);
+  if (!pubKey)
```

Comment:
Normally we seem to add a blank line between clauses.
This would mean adding a blank line before: if (!pubKey)